spec/cucumber/core/test/filters/locations_filter_spec.rb

```diff
@@ -102,57 +102,131 @@ module Core
                   | 1 | 2 |
                   | 3 | 4 |
 
+              Rule: A rule with a background
+                Background: A background rule
+                  Given background
+
+                Scenario: with a rule and background
+                  Given rule a
+
+                Scenario: another with a rule and background
+                  Given rule a
+
+              Rule: A rule without a background
+                Scenario: with a rule and no background
+                  Given rule a
+
+                Scenario: another with a rule and no background
+                  Given rule a
             FEATURE
           end
 
           def test_case_named(name)
             test_cases.find { |c| c.name == name }
           end
 
-          it 'matches the location on a background step to all scenarios' do
+          it 'matches the feature location to all scenarios' do
+            location = Test::Location.new(file, 1)
+            filter = described_class.new([location])
+            compile [doc], receiver, [filter]
+            expect(receiver.test_case_locations).to eq test_cases.map(&:location)
+          end
+
+          it 'matches the feature background location to all scenarios' do
+            location = Test::Location.new(file, 2)
+            filter = described_class.new([location])
+            compile [doc], receiver, [filter]
+            expect(receiver.test_case_locations).to eq test_cases.map(&:location)
+          end
+
+          it 'matches a feature background step location to all scenarios' do
             location = Test::Location.new(file, 3)
             filter = described_class.new([location])
             compile([doc], receiver, [filter])
             expect(receiver.test_case_locations).to eq test_cases.map(&:location)
           end
 
-          it 'matches the precise location of the scenario' do
+          it "matches a rule location (containing a background) to all of the rule's scenarios" do
+            location = Test::Location.new(file, 29)
+            filter = described_class.new([location])
+            compile [doc], receiver, [filter]
+            expect(receiver.test_case_locations).to eq [
```

Comment:
I'm slowly updating the styles of eq to use brackets (We cannot use the rubocop linting yet owing to version).

i.e. `.to eq(value)`